nilearn/datasets/func.py

```diff
@@ -398,8 +398,8 @@ def fetch_adhd(n_subjects=None, data_dir=None, url=None, resume=True,
     Parameters
     ----------
     n_subjects: int, optional
-        The number of subjects to load. If None is given, all the
-        40 subjects are used.
+        The number of subjects to load from maximum of 40 subjects.
+        If default None, only 30 subjects will be loaded.
```

Comment:
> What is puzzling me here is: Is nilearn a tool made for tutorials or for
> studies? People that wants to actually study this dataset may be puzzled by
> this behavior.

But the datasets are only for tutorials / demos. And beside, it's just
like the fetch_haxby, which only downloads 1 subject by default.

> I would be in favor of creating a fetch_tutorial_adhd for this special
> purpose.

How about we keep it simple and don't over engineer things. It's just a
question of changing a default, from n_subjects=None to n_subjects=30.

We've already checked that all the examples work well with only 30
subjects, so we can just stick to the simple solution.